holoviews/plotting/util.py

```diff
@@ -822,8 +822,14 @@ def color_intervals(colors, levels, clip=None, N=255):
         clmin, clmax = clip
         lidx = int(round(N*((clmin-cmin)/interval)))
         uidx = int(round(N*((cmax-clmax)/interval)))
-        cmap = cmap[lidx:N-uidx]
-    return cmap
+        uidx = N-uidx
+        if lidx == uidx:
+            uidx = lidx+1
+        cmap = cmap[lidx:uidx]
+        if clmin == clmax:
+            idx = np.argmin(np.abs(np.array(levels)-clmin))
+            clip = levels[idx: idx+2] if len(levels) > idx+2 else levels[idx-1: idx+1]
+    return cmap, clip
```

Comment:
Now the return value has changed (or at least now includes `clip`) it might be worth updating the docstring...

Reply:
Okay, done.